src/benchmarks/scenes/BenchScene.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <benchmark/benchmark.h>
+
+#include <sofa/simulation/Simulation.h>
+#include <sofa/simulation/Node.h>
+#include <SofaSimulationCommon/SceneLoaderXML.h>
+#include <SofaSimulationGraph/SimpleApi.h>
+#include <SofaSimulationGraph/DAGSimulation.h>
+
+#include <SofaSimulationGraph/init.h>
+#include <SofaComponentAll/initSofaComponentAll.h>
+
+using sofa::simulation::setSimulation;
+
+// Generic benchmark for a scene (timing whole animation)
+// TScene (template argument) needs to implement getRoot(), dt and nbSteps
+template<typename TScene>
+void BM_Scene_bench(benchmark::State& state)
+{
+    sofa::component::initSofaComponentAll();
+
+    sofa::simulation::Simulation* simu = new sofa::simulation::graph::DAGSimulation();
+    setSimulation(simu);
+    sofa::simulation::Node::SPtr root = TScene::getRoot();
+
+    root->init(sofa::core::execparams::defaultInstance());
+
+    for (auto _ : state)
+    {
+        state.ResumeTiming();
+        for (auto i = 0; i < state.range(0); ++i)
```

Comment:
I am not sure to get the logic here. It means you do `state.range(0) * TScene::nbSteps` time steps without reinitializing the scene. Is it relevant?
Why not entirely relying on the benchmark parameters?

Reply:
You are right that I should reset (I dont really trust reinit()...) for every state.
In my mind, what I want is to run `state.range(0)` x nbSteps the simulation (from time=0 every time).
It is to ensure that it will make an avg time.
Here it does not, i will fix that.